### PR TITLE
Add --with-codestart option in create extension

### LINKED ISSUE #26807: GitHub Action CI failing on Windows when the filename is "long"
Windows 😫 
```
  Error: error: unable to create file independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-codestart/java/runtime/src/main/codestarts/quarkus/{extension.id}-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html: Filename too long
```
This path is not that long, I don't see any simple workaround.

Should we enable longpath for Windows in the CI? 
https://github.community/t/filename-too-long-error-for-windows-2019-checkout/16676

_Originally posted by @ia3andy in https://github.com/quarkusio/quarkus/issues/26806#issuecomment-1188775116_

### PR DESCRIPTION
**Pull request by ia3andy**
## A codestart to create codestarts :)

It is now possible to prepate a codestart when creating a new extension (Standalone or Quarkiverse)

- Added to Quarkus CLI (to test it you would need to use the 999-SNAPSHOT CLI):
```bash
quarkus create extension io.quarkiverse.foo:quarkus-foo --with-codestart
```
-  and Quarkus Maven plugin
```bash
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create-extension -DgroupId=io.quarkiverse.foo -DextensionId=foo -DwithCodestart
```

Also Fixes #26807

### CODE REVIEW

**Comment by ia3andy** [comment]
Windows 😫 
```
  Error: error: unable to create file independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-codestart/java/runtime/src/main/codestarts/quarkus/{extension.id}-codestart/base/src/main/resources/META-INF/resources/index.entry.qute.html: Filename too long
```

**Comment by ia3andy** [comment]
@gsmet has this  👆 already happened before? Do we have a way out? Did we consider enabling `git config --system core.longpaths true` https://stackoverflow.com/questions/22575662/filename-too-long-in-git-for-windows?

**Comment by ia3andy** [comment]
I also found this specific to GH actions: https://github.community/t/filename-too-long-error-for-windows-2019-checkout/16676

**Review by aloubyansky — APPROVED** [review]

**Comment by ia3andy** [comment]
Someway I've broken this PR (I just wanted to edit the first commit) I will fix it

**Comment by ia3andy** [comment]
Ok fixed, but I separated the commit

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-codestart/java/integration-tests/src/test/java/{package-name.dir}/it/{class-name-base}CodestartTest.tpl.qute.java [hunk 1]
```diff
@@ -18,7 +17,8 @@
 
     @Test
     void testContent() throws Throwable {
-        codestartTest.checkGeneratedSource("org.acme.{class-name-base}Resource");
+        //codestartTest.checkGeneratedSource("org.acme.SomeClass");
+        //codestartTest.assertThatGeneratedFileMatchSnapshot(Language.JAVA, "\"src/main/resources/some-resource.ext");
```
Comment (aloubyansky) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-codestart/java/integration-tests/src/test/java/{package-name.dir}/it/{class-name-base}CodestartTest.tpl.qute.java:21]: @ia3andy is this supposed to be completed by the user?

**Comment by ia3andy** [comment]
Errors seems unrelated, @gsmet this contains the longpath fix

**Pull request merged**